src/main/java/sirius/biz/storage/layer2/mongo/MongoBlob.java

```diff
@@ -209,6 +215,14 @@ protected void beforeSave() {
         }
 
         updateFilenameFields();
+
+        if (isNew() || isChanged(FILENAME, NORMALIZED_FILENAME, FILE_EXTENSION)) {
+            changed = true;
```

Comment:
maybe we call it createdOrRenamed instead of changed then?

Reply:
What if we extend changed to cover other fields in the future?